src/events/api/serializers.py

```diff
@@ -11,117 +11,168 @@ class PrimarySpeakerSerializer(serializers.Serializer):
+    def get_speakers(self, obj):
+        request = self.context.get('request')
+        return format_speakers_data(request, [obj.host])
 
     class Meta:
         model = SponsoredEvent
+        fields = ["id", "title", "category", "speakers", "is_sponsored"]
+
+
+class TutorialProposalSerializer(serializers.ModelSerializer):
+    speakers = serializers.SerializerMethodField()
+
+    def get_speakers(self, obj):
+        request = self.context.get('request')
+        users = [s.user for s in obj.speakers]
+        return format_speakers_data(request, users, show_details=True)
+
+    class Meta:
+        model = TutorialProposal
         fields = [
-            "slug",
-            "title",
-            "host_name"
+            "title", "category", "language", "python_level",
+            "recording_policy", "abstract", "detailed_description",
+            "slide_link", "slido_embed_link", "speakers",
         ]
 
 
 class TutorialDetailSerializer(serializers.ModelSerializer):
-    title = serializers.CharField(source='proposal.title')
-    date = serializers.DateTimeField(format='%Y-%m-%d', source='begin_time.value')
-    begin_time = serializers.DateTimeField(format='%Y-%m-%d %H:%M:%S', source='begin_time.value')
-    end_time = serializers.DateTimeField(format='%Y-%m-%d %H:%M:%S', source='end_time.value')
-    category = serializers.CharField(source='proposal.category')
-    language = serializers.CharField(source='proposal.language')
-    python_level = serializers.CharField(source='proposal.python_level')
-    abstract = serializers.CharField(source='proposal.abstract')
-    detailed_description = serializers.CharField(source='proposal.detailed_description')
-    slide_link = serializers.CharField(source='proposal.slide_link')
-    slido_embed_link = serializers.CharField(source='proposal.slido_embed_link')
+    proposal = TutorialProposalSerializer()
+
+    def to_representation(self, obj):
+        representation = super().to_representation(obj)
+        return flatten_proposal_field(representation)
+
+    class Meta:
+        model = ProposedTutorialEvent
+        fields = ['proposal', 'begin_time', 'end_time', 'is_remote', 'location']
+
+
+class TutorialListSerializer(serializers.ModelSerializer):
     speakers = serializers.SerializerMethodField()
 
     def get_speakers(self, obj):
-        return [
-            ReturnDict(PrimarySpeakerSerializer(
-                data={'thumbnail_url': i.user.get_thumbnail_url(),
-                      'name': i.user.speaker_name,
-                      'github_profile_url': i.user.github_profile_url,
-                      'twitter_profile_url': i.user.twitter_profile_url,
-                      'facebook_profile_url': i.user.facebook_profile_url}).get_initial(),
-                serializer=PrimarySpeakerSerializer) for i in obj.proposal.speakers]
+        request = self.context.get('request')
+        users = [s.user for s in obj.speakers]
```

Comment:
I think it is more readable to replace the parameter name from users to speakers.